### ISSUE TITLE
notification grouping not working in android api level above 24

### ISSUE THREAD

**Issue by Nethram** [MaikuB/flutter_local_notifications#226]
. I am using flutter_local_notification plugin for showing notification in notification bar. I'm getting different tabs for every notification. So i have used grouping feature of this plugin for grouping notifications. But it seems not working still it is getting as multiple notifications in notification bar. Any help would be appreciated. I tried this in api level above 24 as their document says. Sample code i tried is as below.

String groupKey = 'com.android.example.WORK_EMAIL';

String groupChannelId = 'grouped channel id';

String groupChannelName = 'grouped channel name';

String groupChannelDescription = 'grouped channel description';

AndroidNotificationDetails firstNotificationAndroidSpecifics = new AndroidNotificationDetails( groupChannelId, groupChannelName, groupChannelDescription, importance: Importance.Max, priority: Priority.High, groupKey: groupKey);

NotificationDetails firstNotificationPlatformSpecifics = new NotificationDetails(firstNotificationAndroidSpecifics, null);

await flutterLocalNotificationsPlugin.show(1, 'Alex Faarborg', 'You will not believe...', firstNotificationPlatformSpecifics);

AndroidNotificationDetails secondNotificationAndroidSpecifics = new AndroidNotificationDetails( groupChannelId, groupChannelName, groupChannelDescription, importance: Importance.Max, priority: Priority.High, groupKey: groupKey);

NotificationDetails secondNotificationPlatformSpecifics = new NotificationDetails(secondNotificationAndroidSpecifics, null);

await flutterLocalNotificationsPlugin.show( 2, 'Jeff Chang', 'Please join us to celebrate the...', secondNotificationPlatformSpecifics);

**Comment by MaikuB** [comment]
<img width="376" alt="Screen Shot 2019-05-09 at 6 15 27 pm" src="https://user-images.githubusercontent.com/25263378/57438136-94414380-7286-11e9-998a-0de61be45368.png">

are you perhaps just checking the notifications too quickly before the OS has had a chance to actually group them. the above is a screenshot taken from the example app that runs pretty much the same code you have

**Comment by Nethram** [comment]
Are you using InboxStyleInformation in your code?

**Comment by MaikuB** [comment]
Yes for the summary notification. I've noticed you're missing this in the code you've pasted. That's probably the cause of the issue you're seeing. Note that as mentioned in the readme that there's an example app in this repo that you can run

**Comment by Nethram** [comment]
But for summary notification how can i update it when new push notification comes? I need to show the latest push notification's details as well.

**Comment by MaikuB** [comment]
Took another look and seems like the summary is needed. Could be to do with the note Google has [here](https://developer.android.com/training/notify-user/group#set_a_group_summary) that the behaviour depends on the device. The summary notification is only visible on older devices but believe you should be able to just "update" the notification summary by calling the same code to generate the summary notification but change the details related to the content with the same notification id. Updating a progress notification is done in a similar way by reusing the same notification id

**Comment by Nethram** [comment]
Ok. This is working when the app is in foreground state. But when app is in background or killer, this notification still shows multiple tabs for each notification. I tried provide "tag" key in FCM "notification" section. But can you help me to show grouping even app is is not in foreground.

**Comment by MaikuB** [comment]
I don't know your exact scenario with how you've set everything up since you said you are using FCM and have a notification section, this means that should mean you're sending notification messages and the firebase messaging plugin handles displaying those, not this plugin. As such, you should check the resources on dealing with FCM or ask the on the Flutter repo. You should also look at the table in their readme (https://pub.dev/packages/firebase_messaging) if you haven't already that describes that plugin's behaviour

**Comment by MaikuB** [comment]
Closing this as there doesn't like much for me to add any further since you're dealing with FCM

**Comment by cgalan12** [comment]
Hi, I am having a similar issue. I'm using FCM to send message data to my app (without notification) and using the _onBackgroundMessage to show LocalNotifications using this pluggin. I'm using the groupKey but they don't group in the tray

**Comment by MaikuB** [comment]
You need to setup a summary notification for the grouping to work. I don't know if you've already handled that but using `groupKey` isn't sufficient. If you have issues that relate to using FCM that this isn't something I can help with and you will need to investigate further

**Comment by cgalan12** [comment]
Thank you for answering quickly. You are right, I wasn't add the summary notification. I saw the sample app and there is a comment that says that is required. I copied this comment and I added to the readme in this PR #618 . Thank you for your help.

**Comment by ZotInfo** [comment]
Hello, I'm having trouble grouping notifications too, I'm not using FCM, just Flutter Local Notifications, could you explain better what "summary notification" would be and how can I configure it?

**Comment by jannikbryld** [comment]
Running the example code for grouped notifications, "_showGroupedNotifications", I experience the same behaviour as Nethram, where the notifications are not grouped but rather showed one by one. The summary notification is not shown (as expected on api 24+). Tested on an emulated Pixel 2 and physical Oneplus 7 pro.

Is this a known issue or might I have configured something wrong, although I am using the example code?
